### PR TITLE
Add threat shield page

### PR DESCRIPTION
**Pull request by kevincela**
- Add threat shield view

Card: https://trello.com/c/UbzjFAwP/314-threat-shield-ip-minimal-ui

### CODE REVIEW

File: public/i18n/en/translation.json [hunk 1]
```diff
@@ -1244,6 +1254,37 @@
...
       "no_exception_found": "No exception found"
     },
+    "threat_shield": {
+      "title": "Threat shield",
+      "blocklist": "Blocklist",
+      "allowlist": "Allowlist",
+      "settings": "Settings",
+      "threat_shield_status": "Threat shield status",
+      "threat_shield_status_description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec ultricies lacinia ante at luctus. Suspendisse tristique neque.",
+      "status_tooltip": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec ultricies lacinia ante at luctus. Suspendisse tristique neque.",
```
Comment (gsanchietti) [inline public/i18n/en/translation.json:1264]:
```suggestion
```

File: public/i18n/en/translation.json [hunk 2]
```diff
@@ -1244,6 +1254,39 @@
       "exception_name": "Exception name",
       "no_exception_found": "No exception found"
     },
+    "threat_shield": {
+      "title": "Threat shield",
+      "blocklist": "Blocklist",
+      "allowlist": "Allowlist",
+      "settings": "Settings",
+      "threat_shield_status": "Threat shield status",
+      "threat_shield_status_description": "When enabled, Threat Shield blocks all new connections to and from malicious IP addresses. Blocked connections are logged with the tag 'banIP/{'<'}direction{'>'}-{'<'}zone{'>'}/{'<'}action{'>'}/{'<'}blocklist{'>'}'",
```
Comment (gsanchietti) [inline public/i18n/en/translation.json:1263]:
```suggestion
      "threat_shield_status_description": "When enabled, Threat Shield blocks all new connections to and from malicious IP addresses. Blocked connections are logged with the tag 'banIP/{'<'}direction{'>'}-{'<'}zone{'>'}/{'<'}action{'>'}/{'<'}blocklist{'>'}',",
```

File: public/i18n/en/translation.json [hunk 3]
```diff
@@ -1244,6 +1254,39 @@
...
+      "threat_shield_disabled": "Threat shield disabled",
+      "name": "Name",
+      "type": "Type",
+      "confidence": "Confidence",
+      "no_blacklists_found": "No blacklists found",
+      "go_to_settings": "Go to settings",
+      "community": "Community",
+      "enterprise": "Enterprise",
+      "unknown": "Unknown",
+      "allowlist_description": "Enter a valid IPv4/IPv6 address with optional CIDR notation, a MAC address or a fully qualified host name that are always allowed",
```
Comment (gsanchietti) [inline public/i18n/en/translation.json:1277]:
```suggestion
      "allowlist_description": "Enter a valid IPv4/IPv6 address with optional CIDR notation, a MAC address or a fully qualified host name that are always allowed.",
```

File: public/i18n/en/translation.json [hunk 4]
```diff
@@ -1244,6 +1254,39 @@
...
+      "allowlist_is_empty": "Allowlist is empty",
+      "add_address": "Add address",
+      "edit_address": "Edit address",
+      "address_added": "Address added",
+      "address_edited": "Address edited",
+      "address_deleted": "Address deleted",
+      "address": "Address",
+      "address_tooltip": "Preferably enter an IPv4 or IPv6 (CIDR notation is allowed). You can also enter FQDNs or MAC address in the form xx:xx:xx:xx:xx:xx.",
```
Comment (gsanchietti) [inline public/i18n/en/translation.json:1285]:
```suggestion
      "address_tooltip": "Enter an IPv4 or IPv6,  CIDR notation is allowed. You can also enter an host name or MAC address in the form xx:xx:xx:xx:xx:xx.",
```

File: public/i18n/en/translation.json [hunk 5]
```diff
@@ -268,7 +269,16 @@
     "invalid_otp": "Invalid OTP",
     "cannot_retrieve_qr_code": "Cannot retrieve QR code",
     "cannot_enable_automatic_updates": "Cannot enable automatic updates",
-    "cannot_disable_automatic_updates": "Cannot disable automatic updates"
+    "cannot_disable_automatic_updates": "Cannot disable automatic updates",
+    "cannot_retrieve_threat_shield_settings": "Cannot retrieve threat shield settings",
```
Comment (andre8244) [inline public/i18n/en/translation.json:273]:
```suggestion
    "cannot_retrieve_threat_shield_settings": "Cannot retrieve Threat shield settings",
```

File: public/i18n/en/translation.json [hunk 6]
```diff
@@ -268,7 +269,16 @@
...
+    "cannot_save_threat_shield_settings": "Cannot save threat shield settings",
```
Comment (andre8244) [inline public/i18n/en/translation.json:274]:
```suggestion
    "cannot_save_threat_shield_settings": "Cannot save Threat shield settings",
```

File: public/i18n/en/translation.json [hunk 7]
```diff
@@ -1244,6 +1254,39 @@
...
+    "threat_shield": {
+      "title": "Threat shield",
+      "blocklist": "Blocklist",
+      "allowlist": "Allowlist",
+      "settings": "Settings",
+      "threat_shield_status": "Threat shield status",
+      "threat_shield_status_description": "When enabled, Threat shield blocks all new connections to and from malicious IP addresses. Blocked connections are logged with the tag 'banIP/{'<'}direction{'>'}-{'<'}zone{'>'}/{'<'}action{'>'}/{'<'}blocklist{'>'}'.",
+      "status": "Status",
+      "blocklist_description": "Threat shield keeps you safe by blocking attacks from known malicious IP addresses. These addresses are compiled into blocklists, each with a clear name that tells you its purpose and who maintains it. The confidence score is a value from 1 to 10 that indicates the quality of the list. A higher number means a lower chance of false positives. This score is not available for 'Community' lists.",
+      "blocklist_subscription_description": "Verify if your Enterprise subscription plan includes Threat shield to use this service.",
```
Comment (andre8244) [inline public/i18n/en/translation.json:1266]:
```suggestion
      "blocklist_subscription_description": "Make sure your Enterprise subscription plan includes Threat shield to use this service.",
```

File: public/i18n/en/translation.json [hunk 8]
```diff
@@ -1244,6 +1254,39 @@
...
+      "threat_shield_enabled": "Threat shield enabled",
+      "threat_shield_disabled": "Threat shield disabled",
+      "name": "Name",
+      "type": "Type",
+      "confidence": "Confidence",
+      "no_blacklists_found": "No blacklists found",
```
Comment (andre8244) [inline public/i18n/en/translation.json:1272]:
```suggestion
      "no_blacklists_found": "No blocklists found. Try changing your search filter",
```

File: public/i18n/en/translation.json [hunk 9]
```diff
@@ -1244,6 +1254,39 @@
...
+      "go_to_settings": "Go to settings",
```
Comment (andre8244) [inline public/i18n/en/translation.json:1273]:
```suggestion
      "go_to_settings": "Go to Settings",
```

File: public/i18n/en/translation.json [hunk 10]
```diff
@@ -1244,6 +1254,39 @@
...
+      "community": "Community",
+      "enterprise": "Enterprise",
+      "unknown": "Unknown",
+      "allowlist_description": "Enter a valid IPv4/IPv6 address with optional CIDR notation, a MAC address or a fully qualified host name that are always allowed.",
```
Comment (andre8244) [inline public/i18n/en/translation.json:1277]:
```suggestion
      "allowlist_description": "Enter a valid IPv4/IPv6 address with optional CIDR notation, a MAC address or a fully qualified host name that should always be allowed by Threat shield.",
```

File: src/components/standalone/security/threat_shield/AllowlistTab.vue [hunk 11]
```diff
@@ -0,0 +1,161 @@
...
+      :title="t('error.cannot_retrieve_allowlist')"
+      :description="error.notificationDescription"
+      class="mb-6"
+      kind="error"
+    >
+      <template #details v-if="error.notificationDetails">
+        {{ error.notificationDetails }}
+      </template></NeInlineNotification
+    >
+    <NeSkeleton v-if="loading" :lines="10" />
```
Comment (andre8244) [inline src/components/standalone/security/threat_shield/AllowlistTab.vue:97]:
```suggestion
    <NeSkeleton v-if="loading" :lines="6" :size="lg" />
```
Just to make it more similar to other skeletons :)

File: src/components/standalone/security/threat_shield/AllowlistTab.vue [hunk 12]
```diff
@@ -0,0 +1,161 @@
...
+  error.value.notificationDescription = ''
+  error.value.notificationDetails = ''
+
+  try {
+    loading.value = true
+    allowlist.value = (await ubusCall('ns.threatshield', 'list-allowed')).data.data
+  } catch (err: any) {
+    error.value.notificationDescription = t(getAxiosErrorMessage(err))
+    error.value.notificationDetails = err.toString()
+    fetchError.value = true
```
Comment (andre8244) [inline src/components/standalone/security/threat_shield/AllowlistTab.vue:54]: It seems like this variable is never reset to false, maybe we should do it before the `try`

File: src/components/standalone/security/threat_shield/BlocklistTab.vue [hunk 13]
```diff
@@ -0,0 +1,180 @@
...
+    loading.value = true
+    blocklists.value = (await ubusCall('ns.threatshield', 'list-blocklist')).data.data
+    isThreatShieldEnabled.value = (
+      await ubusCall('ns.threatshield', 'list-settings')
+    ).data.data.enabled
+  } catch (err: any) {
+    error.value.notificationTitle = t('error.cannot_retrieve_blocklists')
+    error.value.notificationDescription = t(getAxiosErrorMessage(err))
+    error.value.notificationDetails = err.toString()
+    fetchError.value = true
```
Comment (andre8244) [inline src/components/standalone/security/threat_shield/BlocklistTab.vue:73]: It seems like this variable is never reset to false, maybe we should do it before the `try`

File: src/components/standalone/security/threat_shield/BlocklistTab.vue [hunk 14]
```diff
@@ -0,0 +1,180 @@
...
+  }
+}
+
+onMounted(() => {
+  fetchBlocklists()
+})
+</script>
+
+<template>
+  <NeSkeleton v-if="loading" :lines="15" />
```
Comment (andre8244) [inline src/components/standalone/security/threat_shield/BlocklistTab.vue:109]:
```suggestion
  <NeSkeleton v-if="loading" :lines="8" :size="lg"/>
```

File: src/components/standalone/security/threat_shield/SettingsTab.vue [hunk 15]
```diff
@@ -0,0 +1,104 @@
...
+  }
+}
+
+onMounted(() => {
+  fetchSettings()
+})
+</script>
+
+<template>
+  <NeSkeleton :lines="10" v-if="loading" />
```
Comment (andre8244) [inline src/components/standalone/security/threat_shield/SettingsTab.vue:70]:
```suggestion
  <NeSkeleton :lines="6" v-if="loading" :size="lg" />
```

File: public/i18n/en/translation.json (hunk 7 above)
Comment (gsanchietti) [inline public/i18n/en/translation.json:1266]:
```suggestion
      "blocklist_subscription_description": "Nethesis and Yoroi blocklist are available only if the machine has a valid Enterprise subscription that includes Threat shield service.",
```

File: src/components/standalone/security/threat_shield/CreateOrEditAddressDrawer.vue [hunk 16]
```diff
@@ -0,0 +1,176 @@
...
+  }
+  if (!addressFormatValidator.valid) {
+    validationErrorBag.value.set('address', addressFormatValidator.errMessage as string)
+    return false
+  }
+
+  return true
+}
+
+async function createOrEditQosInterface() {
```
Comment (andre8244) [inline src/components/standalone/security/threat_shield/CreateOrEditAddressDrawer.vue:57]: Maybe `createOrEditAddress` is a better name :joy: :joy:

**Review by andre8244 — COMMENTED** [review]
Just some minor fixes

**Review by andre8244 — APPROVED** [review]

**Pull request merged**